### DISCUSSION TITLE
Adding an Attachment created by a behavior to an email

### DISCUSSION THREAD

**Discussion by wbtittle** [cerb/cerb-release#54] [Q&A] [answered]
A client wants a spreadsheet sent to them every week with the status of the project. 

I created an Attachment to attach to the email..

I can link to the attachment in the email, but I cannot find the point to ADD the attachment to the email.. 

The link method works, but you have to be logged in to download the file. Can I set the permissions on a file? 

The Behavior Tree
![Screenshot 2023-01-23 at 7 24 40 AM](https://user-images.githubusercontent.com/418631/214087526-d8e9edfa-9a39-492c-9b21-168ff7296550

The Attachment Creation
![Screenshot 2023-01-23 at 8 15 08 AM](https://user-images.githubusercontent.com/418631/214090984-787fa221-33f2-4275-ba26-341335b590a6.png)

The eMail Send..
![Screenshot 2023-01-23 at 8 13 34 AM](https://user-images.githubusercontent.com/418631/214090559-4a41de09-db41-4942-83a3-86b8dbff6c36.png)
.png)

Where is the magical way to attach the attachment to the email?

**Comment by jstanden** [comment, marked as answer]
Hi @wbtittle!

We're in the process of replacing behaviors with [automations](https://cerb.ai/docs/automations/). The main benefit of automations is that you can use scripting and placeholders for any value in all conditions, actions, etc. They also drastically simplify multi-step workflows like [interactions](https://cerb.ai/docs/interactions/). Behaviors by comparison are simpler but much more confined.

What you're looking for is possible in behaviors, but not as intuitive as it should be. If you edit the behavior itself (at the top of your behavior tree in bold), you can add **variables** to it that are available throughout actions. Add a 'List:Attachments' variable. Set it to the value of the attachment you created. Then the 'Send Email' action will show that variable as an option to include.

![CleanShot 2023-01-23 at 15 20 47@2x](https://user-images.githubusercontent.com/63288/214174305-03a1e040-2afc-49d0-a205-506176bfef3f.png)

![CleanShot 2023-01-23 at 15 23 45@2x](https://user-images.githubusercontent.com/63288/214174716-96e14048-3e11-4c5a-be6a-1dc8c4485487.png)

![CleanShot 2023-01-23 at 15 22 49@2x](https://user-images.githubusercontent.com/63288/214174625-6d69ba80-11e1-4ee2-9048-2b10e268f0c5.png)

In automations, we recommend creating a `mail.transactional` [draft](https://cerb.ai/docs/records/types/draft/#params-mailtransactional) record. You can set every property from there, including `file_ids`, GPG signing/encrypting, Markdown formatting, custom fields, etc.

A `mail.transactional` draft will also enter the mail queue so failures are retried, SMTP doesn't block the automation thread, etc. If `is_queued@bool: yes` then it will send immediately. You can also choose a future delivery time.

**Comment by jstanden** [comment]
Here's the guide on sending mail from automations: https://github.com/cerb/cerb-release/discussions/23

**Comment by wbtittle** [comment]
Thank you.

On Mon, Jan 23, 2023 at 3:28 PM Jeff Standen ***@***.***>
wrote:

> Here's the guide on sending mail from automations: #23
> <https://github.com/cerb/cerb-release/discussions/23>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cerb/cerb-release/discussions/54#discussioncomment-4761557>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AADGGR2MR7OOULIIZHSBKOLWT4HZ5ANCNFSM6AAAAAAUEBCYYE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Brad Tittle
Tech Support
